File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>

int solution(int number, int limit, int power) {
    int answer = 0;

    for(int i = 1; i <= number; i++){
        int weapon = 0;

        for(int j = 1; j <= (int) sqrt(i); j++){
            if(i % j == 0) weapon += 2;
            if(j * j == i) weapon -= 1;
        }

        if(limit < weapon) {
            answer += power;
        }else {
            answer += weapon;
        }
        
    }

    return answer;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    
    for i in range(1, number + 1):
        stack = []
        for j in range(1, int(i ** 0.5) + 1):
            if i % j == 0:
                stack.append(j)
                if j ** 2 != i:
                    stack.append(i // j)
        
        if len(stack) > limit:
            answer += power
        else:
            answer += len(stack)
    
    return answer